### PR TITLE
Add new constructor for Histogram and Summary taking rvalue ref of parameters

### PR DESCRIPTION
**Pull request by sjanel**
- Set default values to object fields in header file to limit code duplication as well.
- Mark constructors taking only one parameter as explicit

### CODE REVIEW

File: core/src/summary.cc [hunk 1]
```diff
@@ -20,13 +23,13 @@ void Summary::Observe(const double value) {
 }
 
 ClientMetric Summary::Collect() const {
-  auto metric = ClientMetric{};
```
Comment (gjasny) [inline core/src/summary.cc:23]: Whats wrong with those? We tru to follow the [almost-always-auto](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/) style here.

Comment (sjanel) [reply core/src/summary.cc:23]: Nothing wrong, it's just simpler and not involving `auto` nor assignment. I have not heard about this style, thanks for the link. In the end, I guess it's just a matter of style, personally I am not a fan of this but no problem to revert it to keep style consistent.

File: core/include/prometheus/histogram.h [hunk 2]
```diff
@@ -68,10 +70,10 @@ class PROMETHEUS_CPP_CORE_EXPORT Histogram {
   ClientMetric Collect() const;
 
  private:
-  const BucketBoundaries bucket_boundaries_;
+  BucketBoundaries bucket_boundaries_;
   mutable std::mutex mutex_;
   std::vector<Counter> bucket_counts_;
-  Gauge sum_;
+  Gauge sum_{};
```
Comment (gjasny) [inline core/include/prometheus/histogram.h:76]: I think you could omit the `{}` here. The `sum_` member was only present in the member initializer list to show the order of members.

File: core/src/summary.cc [hunk 3]
```diff
@@ -4,14 +4,17 @@
...
+                 int age_buckets)
     : quantiles_{quantiles},
-      count_{0},
-      sum_{0},
       quantile_values_{quantiles_, max_age, age_buckets} {}
 
-void Summary::Observe(const double value) {
+Summary::Summary(Quantiles&& quantiles, std::chrono::milliseconds max_age,
+                 int age_buckets)
+    : quantiles_(std::move(quantiles)),
```
Comment (gjasny) [inline core/src/summary.cc:14]: Could you make that curly braces as well or will it get ambiguous?

**Review by gjasny — CHANGES_REQUESTED** [review]

File: core/src/summary.cc (hunk 1 above)
Comment (gjasny) [reply core/src/summary.cc:23]: Thanks. I'd like to keep those as-is.

**Review by gjasny — APPROVED** [review]

**Pull request merged**